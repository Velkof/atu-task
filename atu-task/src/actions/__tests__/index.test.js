import configureMockStore from 'redux-mock-store'
import { fetchPosts, fetchPostDetails, fetchPostAuthor } from '../index.js'
import thunk from 'redux-thunk';
import moxios from 'moxios'

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('async actions', () => {
    beforeEach(function () {
      moxios.install()
    });
  
    afterEach(function () {
      moxios.uninstall()
    });
  
    it('fetches posts successfully', () => {
      moxios.stubRequest('https://jsonplaceholder.typicode.com/posts', {
        status: 200,
        response: { data: { 
            posts: [{
            "userId":1,
            "id":1,
            "title":"Example",
            "body":"Testing",
          }] 
        }}
      });
  
      const store = mockStore({});
  
      return store.dispatch(fetchPosts())
        .then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    })

    it('fails fetching posts', () => {
        moxios.stubRequest('https://jsonplaceholder.typicode.com/posts', {
          status: 500,
          response: '[Error: Request failed with status code 500]'
        });
    
        const store = mockStore({});
    
        return store.dispatch(fetchPosts())
          .then(() => {
              expect(store.getActions()).toMatchSnapshot();
          })
      })

      it('fetches post details successfully', () => {
        moxios.stubRequest('https://jsonplaceholder.typicode.com/posts/1', {
          status: 200,
          response: { data: { 
              post: {
              "userId":1,
              "id":1,
              "title":"Example",
              "body":"Testing",
            }
          }}
        });
    
        const store = mockStore({});
    
        return store.dispatch(fetchPostDetails(1))
          .then(() => {
              expect(store.getActions()).toMatchSnapshot();
          })
      })

      it('fails to fetch post details', () => {
        moxios.stubRequest('https://jsonplaceholder.typicode.com/posts/1', {
          status: 500,
          response: '[Error: Request failed with status code 500]'
        });
    
        const store = mockStore({});
    
        return store.dispatch(fetchPostDetails(1))
          .then(() => {
              expect(store.getActions()).toMatchSnapshot();
          })
      })

      it('fetches post author successfully', () => {
        moxios.stubRequest('https://jsonplaceholder.typicode.com/users/1', {
          status: 200,
          response: { data: { 
              user: {
              "id":1,
              "name":"Example name",
            }
          }}
        });
    
        const store = mockStore({});
    
        return store.dispatch(fetchPostAuthor(1))
          .then(() => {
              expect(store.getActions()).toMatchSnapshot();
          })
      })

      it('fails to fetch post author', () => {
        moxios.stubRequest('https://jsonplaceholder.typicode.com/users/1', {
          status: 500,
          response: '[Error: Request failed with status code 500]'
        });
    
        const store = mockStore({});
    
        return store.dispatch(fetchPostAuthor(1))
          .then(() => {
              expect(store.getActions()).toMatchSnapshot();
          })
      })
  });
