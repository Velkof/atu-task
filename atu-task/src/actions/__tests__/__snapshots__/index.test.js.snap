// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async actions fails fetching posts 1`] = `
Array [
  Object {
    "type": "GET_POSTS_REQUEST",
  },
  Object {
    "payload": [Error: Request failed with status code 500],
    "type": "GET_POSTS_FAIL",
  },
]
`;

exports[`async actions fails to fetch post author 1`] = `
Array [
  Object {
    "type": "GET_POST_AUTHOR_REQUEST",
  },
  Object {
    "payload": [Error: Request failed with status code 500],
    "type": "GET_POST_AUTHOR_FAIL",
  },
]
`;

exports[`async actions fails to fetch post details 1`] = `
Array [
  Object {
    "type": "GET_POST_DETAILS_REQUEST",
  },
  Object {
    "payload": [Error: Request failed with status code 500],
    "type": "GET_POST_DETAILS_FAIL",
  },
]
`;

exports[`async actions fetches post author successfully 1`] = `
Array [
  Object {
    "type": "GET_POST_AUTHOR_REQUEST",
  },
  Object {
    "payload": Object {
      "data": Object {
        "user": Object {
          "id": 1,
          "name": "Example name",
        },
      },
    },
    "type": "GET_POST_AUTHOR_SUCCESS",
  },
]
`;

exports[`async actions fetches post details successfully 1`] = `
Array [
  Object {
    "type": "GET_POST_DETAILS_REQUEST",
  },
  Object {
    "type": "GET_POST_AUTHOR_REQUEST",
  },
  Object {
    "payload": Object {
      "data": Object {
        "post": Object {
          "body": "Testing",
          "id": 1,
          "title": "Example",
          "userId": 1,
        },
      },
    },
    "type": "GET_POST_DETAILS_SUCCESS",
  },
]
`;

exports[`async actions fetches posts successfully 1`] = `
Array [
  Object {
    "type": "GET_POSTS_REQUEST",
  },
  Object {
    "payload": Object {
      "data": Object {
        "posts": Array [
          Object {
            "body": "Testing",
            "id": 1,
            "title": "Example",
            "userId": 1,
          },
        ],
      },
    },
    "type": "GET_POSTS_SUCCESS",
  },
]
`;
